File: app/src/main/java/us/shiroyama/wireless/network/SocketReader.java
package us.shiroyama.wireless.network;

import androidx.annotation.NonNull;

public interface SocketReader {
    void read();

    void setOnSuccessCallback(@NonNull OnSuccessCallback onSuccessCallback);

    void setOnFailureCallback(@NonNull OnFailureCallback onFailureCallback);

    interface OnSuccessCallback {
        void onSuccess(@NonNull byte[] message, int length);
    }

    interface OnFailureCallback {
        void onFailure(@NonNull Exception exception);
    }
}

File: app/src/main/java/us/shiroyama/wireless/network/SocketReader.kt
package us.shiroyama.wireless.network

interface SocketReader {
    fun read()
    fun setOnSuccessCallback(onSuccessCallback: OnSuccessCallback)
    fun setOnFailureCallback(onFailureCallback: OnFailureCallback)
    interface OnSuccessCallback {
        fun onSuccess(message: ByteArray, length: Int)
    }

    interface OnFailureCallback {
        fun onFailure(exception: Exception)
    }
}